modules/userdirectory-brightspace/src/main/java/org/opencastproject/userdirectory/brightspace/BrightspaceUserProviderFactory.java

```diff
@@ -0,0 +1,246 @@
+   */
+  @Override
+  public void updated(String pid, Dictionary properties) throws ConfigurationException {
+    logger.debug("updated BrightspaceUserProviderFactory");
+    String organization = (String) properties.get(ORGANIZATION_KEY);
+    String urlStr = (String) properties.get(BRIGHTSPACE_URL);
+    String systemUserId = (String) properties.get(BRIGHTSPACE_USER_ID);
+    String systemUserKey = (String) properties.get(BRIGHTSPACE_USER_KEY);
+    final String applicationId = (String) properties.get(BRIGHTSPACE_APP_ID);
+    final String applicationKey = (String) properties.get(BRIGHTSPACE_APP_KEY);
+
+    int cacheSize = parseCacheSizeProperty(properties);
+    int cacheExpiration = parseCacheExpirationProperty(properties);
+
+    validateOrganization(organization);
+    validateUrl(urlStr);
+    validateSystemUserKey(systemUserKey);
+    validateSystemUserId(systemUserId);
+    validateApplicationId(applicationId);
+    validateApplicationKey(applicationKey);
+
+    ServiceRegistration existingRegistration = this.providerRegistrations.remove(pid);
+    if (existingRegistration != null) {
+      existingRegistration.unregister();
+    }
+
+    Organization org;
+    try {
+      org = this.orgDirectory.getOrganization(organization);
+    } catch (NotFoundException nfe) {
+      logger.warn("Organization {} not found!", organization);
+      throw new ConfigurationException(ORGANIZATION_KEY, "not found");
+    }
+
+    logger.debug("creating new MoodleUserProviderInstance for pid=" + pid);
+
+
+    BrightspaceUserProviderInstance provider = new BrightspaceUserProviderInstance(pid,
+            new BrightspaceClientImpl(urlStr, applicationId, applicationKey, systemUserId, systemUserKey), org, cacheSize,
+            cacheExpiration);
+    this.providerRegistrations
+            .put(pid, this.bundleContext.registerService(UserProvider.class.getName(), provider, null));
+    this.providerRegistrations
+            .put(pid, this.bundleContext.registerService(RoleProvider.class.getName(), provider, null));
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see org.osgi.service.cm.ManagedServiceFactory#deleted(java.lang.String)
+   */
+  @Override
+  public void deleted(String pid) {
+    logger.debug("delete BrightspaceUserProviderInstance for pid=" + pid);
+    ServiceRegistration registration = providerRegistrations.remove(pid);
+    if (registration != null) {
+      registration.unregister();
+
+      try {
+        ManagementFactory.getPlatformMBeanServer().unregisterMBean(getObjectName(pid));
+      } catch (Exception var4) {
+        logger.warn("Unable to unregister mbean for pid='{}': {}", pid, var4.getMessage());
+      }
+    }
+  }
+
+  private int parseCacheExpirationProperty(Dictionary properties) {
+    try {
+      if (properties.get(CACHE_EXPIRATION) != null) {
+        return Integer.parseInt(properties.get(CACHE_EXPIRATION).toString());
+      }
+    } catch (NumberFormatException nfe) {
```

Comment:
My general suggestion for configuration parsing would be to rather fail early if a configuration is broken than to fall back to a default value. That way, as a system administrator you will notice that problem immediately and can fix it instead of accidentally running with an incorrect value for whatever time ;-)

In general, I would also suggest using common-lang3's [NumberUtils::toLong(String str, long defaultValue)](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/math/NumberUtils.html#toLong-java.lang.String-long-). Makes your life much easier and basically any Opencast module uses it anyway.

Reply:
I mean [toInt(String str, int defaultValue)](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/math/NumberUtils.html#toInt-java.lang.String-int-)